### ISSUE TITLE
8$: disallow commas in class fields

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#835]
```javascript
class Potato {
  sprouts = [], age = 0  // error: , should be ; instead
}
console.log(new Potato().age)
```

**Comment by metal-oopa** [comment]
Hi @strager! I want to work on this issue. 
Sending a PR shortly.

**Comment by strager** [comment]
@metal-oopa Any updates? Do you need help making the PR?

**Comment by strager** [comment]
@metal-oopa Someone else wants to work on this task. Did you finish it, or can they pick up this task?

**Comment by metal-oopa** [comment]
> @metal-oopa Someone else wants to work on this task. Did you finish it, or can they pick up this task?

Yeah! They can work. 
I forgot about this issue. Sorry!

**Comment by cmstoddard** [comment]
gimme plox

**Comment by cmstoddard** [comment]
@strager can you give me an idea of what files I should be looking at to start?

**Comment by strager** [comment]
@cmstoddard Tests are in test/test-parse-class.cpp. Add the new error type in src/quick-lint-js/fe/diagnostic-types.h. Implement your fix in src/quick-lint-js/fe/parse-class.cpp.

quick-lint-js reports E0223 for the following code:
```javascript
class Potato {
  sprouts = [] age = 0  // error: , should be ; instead
}
console.log(new Potato().age)
```
According to src/quick-lint-js/fe/diagnostic-types.h, E0223 corresponds to the class diag_missing_semicolon_after_field. That diag class is reported using the following line which is repeated a few times in src/quick-lint-js/fe/parse-class.cpp: `p->consume_semicolon<diag_missing_semicolon_after_field>();` So I think parse-class.cpp is where we wanna look.

I think the issue is that the comma in `[], age = 0` is treated as the comma binary operator. We can disable parsing commas as binary operators by setting `.commas=false` when calling `parse_and_visit_expression` in `parse_field_initializer` (in parse-class.cpp).
